src/modules/mc_pos_control/mc_pos_control_main.cpp

```diff
@@ -2708,7 +2706,7 @@ MulticopterPositionControl::calculate_thrust_setpoint()
 	 * project the desired thrust force vector F onto the real vehicle's thrust axis in NED:
 	 * body thrust axis [0,0,-1]' rotated by R is: R*[0,0,-1]' = -R_z */
 	matrix::Vector3f R_z(_R(0, 2), _R(1, 2), _R(2, 2));
-	matrix::Vector3f F(thrust_sp.data);
+	matrix::Vector3f F(thrust_sp);
```

Comment:
The extra variable F was only for interface conversion to matrix library. You can delete this line and just write the next one: `float thrust_body_z = thrust_sp.dot(-R_z);`

Reply:
Thanks, updated.